src/parser/mage/fire/modules/talents/FromTheAshes.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import Analyzer, { SELECTED_PLAYER, Options } from 'parser/core/Analyzer';
+import Events, { DamageEvent } from 'parser/core/Events';
+import { formatNumber } from 'common/format';
+import Statistic from 'interface/statistics/Statistic';
+import STATISTIC_CATEGORY from 'interface/others/STATISTIC_CATEGORY';
+import BoringSpellValueText from 'interface/statistics/components/BoringSpellValueText';
+import HIT_TYPES from 'game/HIT_TYPES';
+import SpellUsable from 'parser/shared/modules/SpellUsable';
+import { FIRE_DIRECT_DAMAGE_SPELLS, PHOENIX_FLAMES_MAX_CHARGES } from 'parser/mage/shared/constants';
+
+const MS_REDUCTION_PER_STACK = 1000;
+
+class FromTheAshes extends Analyzer {
+  static dependencies = {
+    spellUsable: SpellUsable,
+  };
+  protected spellUsable!: SpellUsable;
+
+  cooldownReduction = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.FROM_THE_ASHES_TALENT.id);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(FIRE_DIRECT_DAMAGE_SPELLS), this.onCritDamage);
+  }
+
+  //Look for crit damage events to reduce the cooldown on Kindling
+  onCritDamage(event: DamageEvent) {
+    if (!this.spellUsable.isOnCooldown(SPELLS.PHOENIX_FLAMES.id) || event.hitType !== HIT_TYPES.CRIT) {
+      return;
+    }
+    const chargesOnCD = PHOENIX_FLAMES_MAX_CHARGES - this.spellUsable.chargesAvailable(SPELLS.PHOENIX_FLAMES.id);
+    this.cooldownReduction += this.spellUsable.reduceCooldown(SPELLS.PHOENIX_FLAMES, MS_REDUCTION_PER_STACK * chargesOnCD);
```

Comment:
This should be chargesOffCD, right?

Reply:
No. From the Ashes gives you CDR for every charge that is on CD. so this is the max charges (2) - the number available

Reply:
![image](https://user-images.githubusercontent.com/29204244/96800844-9c535b80-1406-11eb-8f53-f065e4785731.png)
I just looked it up on wowhead and it says off cooldown?

Reply:
woops, i misread it

Reply:
That said - is it really per off CD charge for the CDR, and not only for the mastery? Blizzard tooltips fine at work here 😄

Reply:
actually, re-reading it again, i suppose its 1 sec CDR regardless of charges, and the mastery is based on charges